contracts/zero-ex/contracts/src/external/Puppet.sol

```diff
@@ -0,0 +1,175 @@
+/*
+
+  Copyright 2020 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.6.5;
+pragma experimental ABIEncoderV2;
+
+import "@0x/contracts-utils/contracts/src/v06/errors/LibRichErrorsV06.sol";
+import "@0x/contracts-utils/contracts/src/v06/errors/LibOwnableRichErrorsV06.sol";
+import "../errors/LibPuppetRichErrors.sol";
+import "./IPuppet.sol";
+
+
+/// @dev A contract that can execute arbitrary calls from its owner.
+contract Puppet is
+    IPuppet
+{
+    // solhint-disable no-unused-vars,indent,no-empty-blocks
+    using LibRichErrorsV06 for bytes;
+
+    // solhint-disable
+    /// @dev Store the owner/deployer as an immutable to make this contract stateless.
+    address public override immutable owner;
```

Comment:
Features don't have owners. The proxy does. Using an immutable would mean you couldn't transfer ownership, which would make migrations awkward since they rely on this mechanic.

Reply:
Puppet isn't a feature, if that's a point of confusion. It's a standalone contract deployed by the TransformERC20 feature.

Reply:
Ah, gotcha. Makes sense for the proxy. If we do eventually create `onlyOwner` functions in features I could be in favor of this pattern, but suppose we'll x that bridge when we come to it 🤓.